runtime/lua/vim/filetype.lua

```diff
@@ -1680,24 +1680,27 @@ local pattern = {
       return 'git'
     end
   end,
-  ['.*%.[Cc][Ff][Gg]'] = function(path, bufnr)
-    require('vim.filetype.detect').cfg(bufnr)
-  end,
+  ['.*%.[Cc][Ff][Gg]'] = {
+    function(path, bufnr)
+      return require('vim.filetype.detect').cfg(bufnr)
+    end,
+    { priority = -1 },
```

Comment:
Can you leave a comment why this is needed (even if it's just reference which other filetypes this pattern "competes" with, it's useful knowledge to avoid a Chesterton's Fence situation).